lib/authentication/view/edit_profile_page.dart

```diff
@@ -300,9 +298,9 @@ class _EditProfilePageState extends State<EditProfilePage> {
                           child: _changeEmailConfirmationDialog(context))
                       .then((value) => result = value ?? false);
                 }
-                if (image != null) {
+                if (uploadedImage != null) {
                   result = await authProvider.uploadProfilePicture(
-                      uploadImage, context);
+                      imageAsPNG, context);
```

Comment:
There's a bad race condition here. If I press 'Save' before `convertToPNG` has time to complete, it will try to upload `null` as a profile picture. You need to either call (and `await`) `convertToPNG` directly here in the save callback, or use a mutex to make sure the conversion actually finished.